avocado/core/runner.py

```diff
@@ -395,8 +394,12 @@ def sigtstp_handler(signum, frame):     # pylint: disable=W0613
             self.job.log.debug('')
 
         self.result.check_test(test_state)
-        if not status.mapping[test_state['status']]:
-            failures.append(test_state['name'])
+        if test_state['fail_class'] == 'TestTimeoutSkip':
+            # The test was skipped but the job was interrupted due
+            # to the timeout.
+            summary.append('INTERRUPTED')
```

Comment:
Actually it'd be better to detect interrupted tests by `test_state['status'] == "INTERRUPTED"`. Users might add custom plugins with different tests supporting the status to be `INTERRUPTED`.